s3-source-connector/src/integration-test/java/io/aiven/kafka/connect/s3/source/IntegrationTest.java

```diff
@@ -332,15 +357,17 @@ private static ByteArrayOutputStream getAvroRecord(final Schema schema, final in
         return outputStream;
     }
 
-    private static void writeToS3(final String topicName, final byte[] testDataBytes, final String partitionId)
-            throws IOException {
+    private static void writeToS3(final String topicName, final byte[] testDataBytes, final String partitionId,
+            final Set<String> offsetKeys) throws IOException {
         final String filePrefix = topicName + "-" + partitionId + "-" + System.currentTimeMillis();
         final String fileSuffix = ".txt";
 
         final Path testFilePath = File.createTempFile(filePrefix, fileSuffix).toPath();
+        final String objectKey = filePrefix + fileSuffix;
         try {
             Files.write(testFilePath, testDataBytes);
-            saveToS3(TEST_BUCKET_NAME, "", filePrefix + fileSuffix, testFilePath.toFile());
+            saveToS3(TEST_BUCKET_NAME, "", objectKey, testFilePath.toFile());
+            offsetKeys.add(OBJECT_KEY + SEPARATOR + objectKey);
         } finally {
             Files.delete(testFilePath);
         }
```

Comment:
GitHub does not allow to post suggestions for deleted lines, so will try to post as comment.
After applying another suggestion and getting rid of `offsetKeys` arguments it could be possible to make this method look like this, that seems to be a lot more concise and we would avoid creating temporary files and we could get rid of another method - `saveToS3` that is used only here and IMO is redundant.
```
private static String writeToS3(final String topicName, final byte[] testDataBytes, final String partitionId) {
        final String objectKey = topicName + "-" + partitionId + "-" + System.currentTimeMillis() + ".txt";
        final PutObjectRequest request = new PutObjectRequest(TEST_BUCKET_NAME, objectKey, new ByteArrayInputStream(testDataBytes), new ObjectMetadata());
        s3Client.putObject(request);
        return OBJECT_KEY + SEPARATOR + objectKey;
    }
```

Reply:
As discussed will be addressed in another PR